### PR TITLE
Industrial Sector

### PR DESCRIPTION
**Pull request by JosephCarrington**
#### Summary

SUMMARY: Content "Adds industrial sectors to map"

#### Purpose of change

Adds city-like generation for industrial areas. These areas may appear on the outskirts of cities, or on their own far away from cities. These areas will be well stocker with raw resources and food, but will have mid-game level enemies. These areas will also have heavy machinery that the player may be able to use to fabricate otherwise un-craftatable items. At present, there are cities, larger overmap specials, and labs. Cities can span from early to mid-game (or late game if a tank shows up). Larger overmap specials can be anywhere from early to late. The goal of industrial sectors is to provide dependable mid-game content.

Many of the ideas come from this Reddit thread: 
https://www.reddit.com/r/cataclysmdda/comments/bdg1oq/industrial_zones_and_industy_in_cdda/

### CODE REVIEW

File: data/json/mapgen/industrial/slaughterhouse.json [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    {
+      "id": "f_hanging_meathook",
+      "type": "furniture",
+      "name": "hanging meathook",
+      "symbol": "g",
+      "required_str": -1,
+      "move_cost_mod": 2,
+      "color": "light_gray",
+      "deconstruct": { "items": [ { "item": "grip_hook", "count": 1 }, { "item": "chain", "count": 1 } ] },
+      "flags": [ "TRANSPARENT", "SLAUGHTER_EQ" ]
```
Comment (anothersimulacrum) [inline data/json/mapgen/industrial/slaughterhouse.json]: Furniture and itemgroups that are added should go into separate files, such as furniture.json or one of the itemgroups files.
Also, 
```suggestion
      "flags": [ "TRANSPARENT", "BUTCHER_EQ" ]
```

**Comment by Night-Pryanik** [comment]
Please fill in the OP's sections with text describing your PR in detail.

File: data/json/items/melee.json [hunk 2]
```diff
@@ -2258,5 +2258,24 @@
     "techniques": "RAPID",
     "qualities": [ [ "CUT", 1 ], [ "CUT_FINE", 1 ], [ "BUTCHER", 7 ] ],
     "flags": [ "STAB", "SHEATH_KNIFE" ]
+  },
+  {
+    "type": "TOOL",
+    "category": "weapons",
+    "id": "grip_hook",
+    "name": "grip hook",
+    "description": "A simple steel meathook with a simple steel handle. Makes a passable melee weapon.",
```
Comment (Night-Pryanik) [inline data/json/items/melee.json]:
```suggestion
    "description": "A simple steel meathook with a simple steel handle.  Makes a passable melee weapon.",
```

Comment (JosephCarrington) [reply data/json/items/melee.json]: It's a draft PR which I will be working on for a while. I thought I'd start it here, but it's nowhere near finished, including the description of the PR

Comment (JosephCarrington) [reply data/json/items/melee.json]: I can understand how from a maintainer however that a PR without a good description is difficult to parse, I'll make a decent description as my next step.

**Comment by I-am-Erk** [comment]
Besides a description, also don't forget to add roofs to your buildings.

I'd recommend just adding some maps for now, and looking into making an "industrial zone" later. That's a lot of changes for one PR

**Comment by I-am-Erk** [comment]
Besides a description, also don't forget to add roofs to your buildings.

I'd recommend just adding some maps for now, and looking into making an "industrial zone" later. That's a lot of changes for one PR

**Comment by JosephCarrington** [comment]
@I-am-Erk I am confused. If I've done nothing else, it's add roofs to my buildings.

**Comment by I-am-Erk** [comment]
Don't be confused, that is a semiautomated reminder, not a response after reviewing the code.

**Comment by JosephCarrington** [comment]
@I-am-Erk do you think it's a better idea to make a bunch of smaller PR's based on the items in these projects: https://github.com/JosephCarrington/Cataclysm-DDA/projects ?

**Comment by I-am-Erk** [comment]
Yes it is.

What you have going on here is a reasonably sized PR, you don't have to do each step separately.... but you should avoid doing a single large PR that involves mapping, items, monster additions, and code changes all at once. I'd say adding items and maps is plenty for one PR. Monsters is good for another. Then see if you can work out how to do an industrial sector using c++

**Comment by JosephCarrington** [comment]
So would it be better to add all the buildings at once? Or a few at a time? Or one at a time? Thanks for you help BTW

**Comment by I-am-Erk** [comment]
As a a general rule of thumb it's nice if you can keep a PR to under 1000 lines changed,  especially if we're still getting to know you. You can get away with more for some types of PR but since this is straight content addition it's a good amount.

In general more small PRs are preferred to few large PRs

**Comment by JosephCarrington** [comment]
Thanks!

**Pull request closed without merging**